travis: deploy snapshots on every commit to master

tomerweller opened the pull request:
modify travis configuration to deploy snapshots on every commit to master
exclude bifrost from artifact build

bartekn commented on support/scripts/build_release_artifacts/main.go:
```diff
@@ -19,6 +19,7 @@ import (
 )
 
 var extractBinName = regexp.MustCompile(`^(?P<bin>[a-z-]+)-(?P<tag>.+)$`)
+var excludePkgs = []string{"services/bifrost"}
```
In my comment: https://github.com/stellar/go/issues/202#issuecomment-350781950 I meant that we should skip cross-compliation (so building binaries for Windows and Mac). Linux binary should compile just fine in Travis.

tomerweller replied:
Got it.

bartekn commented on .travis.yml:
```diff
@@ -27,15 +35,18 @@ notifications:
     on_success: change
     on_failure: always
-    secure: Uhx8H2RFNsJ9GknDItSwhMOl5K3HTGNjifJwkLUiRio4JH4PzrkHDOzCOpEzrboviVF+Rch9VsC7HchIY2uyqeB9UW63dCMPeaYSs7rfNCmW8bTcGydeVOe0HV18mpM/4yPy/gsw5Zq+eGq7+p9NZtwkOHeQ/YtoSFcWbQQYs3CPZpBUP95KW1fGTn8OIW6nKkVA0Kyr6aXJFZB9AHIg24R9l2uraHdjg5EUJJO/4yR0GWC1pLTfUCqUt9mXPeGSQOhhvR+pCKSljY3pB6oex2hKH04u3U1iRzOBuKMPmMrcyJDzvVRkCaCn+0LPGZ3vgDDcDdcqYxEk1a1n27wzjqYlDeH4xaYTZGpB18LiX6MAm84+iviL5JILIMQ4NcqVNB2 ...
+    secure: Uhx8H2RFNsJ9GknDItSwhMOl5K3HTGNjifJwkLUiRio4JH4PzrkHDOzCOpEzrborfNCmW8bTcGydeVOe0HV18mpM/4yPy/gsw5Zq+eGq7+p9NZtwkOHeQ/YtoSFcWbQQYs3CPZpBUPl2uraHdjg5EUJJO/4yR0GWC1pLTfUCqUt9mXPeGSQOhhvR+pCKSljY3pB6oex2hKH04u3U1iRzOEk1a1n27wzjqYlDeH4xaYTZGpB18LiX6MAm84+iviL5JILIMQ4NcqVNB2L37zl1UW+BMQUSH6+zwztzxhJOz6PAE1aA3O+BI1xuSft6sGY07Z0Q9iKO/uCWIHbsfLso16kIubiHzwjSAz90tbCc8VcwWPXCg8mRFCCuXgr4dev7I+v ...
```
What have changed here?

tomerweller replied:
Mistake. correcting

bartekn commented on .travis.yml:
```diff
@@ -27,15 +35,18 @@ notifications:
+  name: Snapshots $(date --utc +'%F-%T')
+  on:
+    branch: master
+    go: '1.8'
```
OK, it will deploy snapshots on every commit to master but doesn't this change break the previous `tags: true` releases?

tomerweller replied:
Did that ever work? What's the desired behavior here? Do we want to build all artifacts for any new tag?

bartekn replied:
I think it worked. So the process was like this: create a branch with the following format `{app-name}-{version}` (example: https://github.com/stellar/go/releases/tag/federation-v0.2.1) and it will get deployed to GitHub releases with the given name.

tomerweller replied:
I misunderstood this. Will come up with a different flow.

alpe commented:
I don't know about your release pipeline but I found it often very critical to deploy on a git push rather than have the deployment fully automated and trigger it manually on demand.
Some things to consider (you don't need to answer them here):
- Is every update on master a serious release candidate? (expensive/ time consuming steps afterwards)
- do you deploy to a new unique or a shared environment?
- when shared, do you lock / restrict access to prevent concurrent access/ updates?
- do you do manual QA on that environment? If yes it can be very frustrating to have it redeployed while somebody is testing. Also the tester should know which version they are testing (lock environment?)
- can you deploy branches automated as well? (demos/ tests/ etc)

tomerweller commented:
@bartekn I've fixed this to restore tagged versions and bifrost linux build.
Notes: 
1. There are two deploy targets provided: snapshots and tagged versions. Both of them go to the github releases page, but with different triggers. 
2. I extracted the git tag manipulation commands into an external bash script, which checks what type of deploy triggered it. (Unfortunately, you can't have multiple `before_deploy` scripts). 
3. I excluded non-linux bifrost builds manually. This is extremely hacky but I think anything more elegant would require more refactoring than is healthy for this scope.

tomerweller commented:
@alpe sorry, this might be misleading. By *deploying* I'm actually referring to building the binary artifacts and posting them on the github releases page ("deploying" to github). This is travis-ci terminology.

This does not actually deploy these services anywhere.

alpe commented:
@tomerweller thanks for the clarification 💐
